Feature

abhishekpatel946 opened the pull request:
All Request Change are done.

abhishekpatel946 committed:
Initialize project using Create React App

abhishekpatel946 committed:
initial commit

host the basic layout for reminder_app built with materialUI with Hooks

abhishekpatel946 committed:
fix: resolve conflict from unrelated histories

abhishekpatel946 committed:
fix: change after review, only past reminders present

abhishekpatel946 committed:
fix: make it more readable and consice

break into components

abhishekpatel946 committed:
docs: Update the ReadME

add Pending task as a feature in ReadME file

abhishekpatel946 committed:
fix: filter out Past reminder & Upcoming reminder

abhishekpatel946 committed:
fix: filter function

past reminder & future reminder now fixed using simple if condition, but its not accurate

abhishekpatel946 committed:
docs: add demo images in ReadME

abhishekpatel946 committed:
docs: update images in ReadME

abhishekpatel946 committed:
feat: add logo with name & modal

added web-logo with app name and madal for form;s (addUser & updateUser)

abhishekpatel946 committed:
fix: Date & Time with Moment, add Float button to addReminder

change date&time with moment lib, add floaing-right button, change the flow in addReminder & updateReminder

abhishekpatel946 committed:
docs: Update the ReadMe

The pending task is clear

abhishekpatel946 committed:
fix: style in Table & Floating btn

align left the title in table, and use margin inside the floating button area

abhishekpatel946 committed:
Merge branch 'feature' of https://github.com/abhishekpatel946/React-Assignment-10 into feature

abhishekpatel946 committed:
docs: update assest in ReadME

hrathi22 commented on src/App.scss:
```diff
@@ -0,0 +1,25 @@
+@import './SCSS/colorStack';
```
Dont use CAPITAL case for folder name

hrathi22 commented on src/container/Alert/Alert.jsx:
```diff
@@ -0,0 +1,5 @@
+import MuiAlert from '@material-ui/lab/Alert';
+
+export function Alert(props) {
```
Use arrow function here

hrathi22 commented on src/container/Alert/SnackbarMui.jsx:
```diff
@@ -0,0 +1,40 @@
+import PropTypes from 'prop-types';
+import { Alert } from './Alert';
+
+const AlertMui = (props) => {
```
The component name should be meaningful

hrathi22 commented on src/container/Form/AddReminder.jsx:
```diff
@@ -0,0 +1,84 @@
+        value={reminder.title}
+        onChange={handleInputChange}
+      />
+      <br />
```
Dont use `<br>` use grid layout from material UI

hrathi22 commented on src/container/Form/AddReminder.jsx:
```diff
@@ -0,0 +1,84 @@
+
+  return (
+    <form id='addReminderFormId'>
+      <InputText
```
can use https://material-ui.com/api/form-group/

hrathi22 commented on src/container/Form/AddReminder.jsx:
```diff
@@ -0,0 +1,84 @@
+  // destructring props
+  const { addNewReminder } = props;
+
+  const [reminder, setReminder] = useState([]);
```
Type mismatch, Why assigning array as initial value here?

hrathi22 commented on src/container/Form/EditReminder.jsx:
```diff
@@ -0,0 +1,108 @@
+  };
+
+  return (
+    <form id='editReminderFormId'>
```
Why two diff form for add and edit ?

hrathi22 commented on src/container/Home/Home.jsx:
```diff
@@ -0,0 +1,280 @@
+
+  // write a cron job for filter(past & future) the all the reminders
+  useEffect(() => {
+    let todayDate = new Date();
```
too much logic in a single component, can we split it?

hrathi22 commented on src/container/Table/ReminderTable.jsx:
```diff
@@ -0,0 +1,109 @@
+      </AppBar>
+
+      {/* all reminders */}
+      <TabPanel value={value} index={0}>
```
Can we have one json for all tab list and loop it.. can reuse the code

abhishekpatel946 replied:
currently I'm working on it.

hrathi22 commented on src/container/Table/a11yProps.jsx:
```diff
@@ -0,0 +1,6 @@
+export function a11yProps(index) {
```
Check name of the file, wrong way to write the file name

hrathi22 commented on src/logo.svg:
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 841.9 595.3"><g fill="#61DAFB"><path d="M666.3 296.5c0-32.5-40.7-63.3-103.1-82.4 14.4-63.6 8-114.2-20.2-130.4-6.5-3.8-14.1-5.6-22.4-5.6v22.3c4.6 0 8.3.9 11.4 2.6 13.6 7.8 19.5 37.5 14.9 75.7-1.1 9.4-2.9 19.3-5.1 29.4-19.6-4.8-41-8.5-63.5-10.9-13.5-18.5-27.5-35.3-41.6-50 32.6-30.3 63.2-46.9 84-46.9V78c-27.5 0-63.5 19.6-99.9 53.6-36.4-33.8-72.4-5 ...
```
no new line at EOF

hrathi22 reviewed (changes requested):
Please work on given comments

abhishekpatel946 committed:
fix: change requested code

The entire change request are listed in the PR. https://github.com/abhishekpatel946/React-Assignment-10/pull/1

abhishekpatel946 committed:
fix: Tabs are usable with JSON data

tabs are reusable with JSON data to render the table with map

abhishekpatel946 committed:
feat: add routes with SignIn & SignUp form & 404 Page

abhishekpatel946 committed:
feat: login with firebase using privateRoute

add routes using firebase auth & db with NavBar, SignIn, SignUp, Error404, ResetPassword with ContextAPI pages are added and authentication working now

abhishekpatel946 committed:
fix: AppBar, dbService

fix the broken UI components, and add the dbService with firestore get the data successfully as of now and still working on setter & getter with individual users with watchers

abhishekpatel946 committed:
initial code pushed for review

abhishekpatel946 committed:
fix: CRUD with firestore is completed,

abhishekpatel946 committed:
fix: setting up newUser and store the info into database

abhishekpatel946 committed:
feat: add lazyLoading with spinner after login

abhishekpatel946 committed:
docs: Update the ReadMe for new feature showcase

abhishekpatel946 committed:
fix: get rid up the warnings from Components

fix the descendant warnings from the components & mistypos & handle the proptypes as well

abhishekpatel946 committed:
fix: handle the warnings after guidance. (Table, TabPanel)

abhishekpatel946 committed:
docs: add pendng Features are implement ASAP.

abhishekpatel946 committed:
feat: initial setup Redux-toolkit, Redux-saga, logger

create redux(toolkit), saga(as middleware) and logger(as middleware) and connect each of them

abhishekpatel946 committed:
fix: Getting stuck in Redux-Saga structure.

abhishekpatel946 committed:
fix: log the fetch data from API in getReminder.js

abhishekpatel946 committed:
fix: add new Redux-saga structure

abhishekpatel946 committed:
fix: commit for code review, after getting a problem.

abhishekpatel946 commented on src/helper/Redux/sagas/reminder/reminder.saga.js:
```diff
@@ -1,20 +1,52 @@
 function* fetchReminder() {
   try {
-    const reminders = yield call(reminderService.getAllReminders);
+    const reminders = yield call(GetAllReminders);
+    console.log(`reminders =>> ${reminders}`); // reminders =>> [object Object]
+
+    const resultDoc = GetDataFromSnapshot(reminders);
+
+    resultDoc
+      .then((data) => {
+        console.log(`resultDoc =>> ${resultDoc}`); // resultDoc =>> [object Promise]
+        console.log(`data =>> ${data}`);
+        // data =>> function () {
+        //   i.Zl(), r.cs.ws(function () {
+        //     return Pr(r.q_, o);
+        //   });
+        // }
+
+        // data.json();  // Unhandled Rejection (TypeError): data.json is not a function
+      })
+      .then((response) => console.log(`response =>> ${response}`)); // response =>> undefined
```
The problem arises here when you see a log with each output that I put here for simplicity.
I expected the data as an object of JSON `data.json()` will easily be converted and get the result. But here will be something like callback fn this will be I didn't get the properly

abhishekpatel946 commented on src/helper/Redux/service/reminders.service.js:
```diff
@@ -1,36 +1,11 @@
-const reminderService = Service();
-
-export default reminderService;
+  return await db
+    .collection('users')
+    .doc(currentUser.uid)
+    .collection('reminders');
+};
```
here we will successfully return the object of collection. This is run perfectly no issues with this fn.

abhishekpatel946 commented on src/helper/Utils/GetDataFromSnapshot.js:
```diff
@@ -0,0 +1,17 @@
+export const GetDataFromSnapshot = async (snapshot) => {
+  const data = [];
+  return await snapshot.onSnapshot(
+    (docs) => {
+      const currentState = [];
+      docs.forEach((doc) => {
+        console.log(doc.data()); // individual doc object
+        currentState.push(doc.data());
+      });
+      data.push(currentState);
+      console.log(currentState); // all the stored docs
+    },
+    (error) => {
+      console.log(error);
+    }
+  );
+};
```
This is `utility` to `GetDataFromSnapshot` we will return from our collection, and dive into individual document loop through it and store into an array. And once we will get the all data, then we return back to our reducer fn.

abhishekpatel946 commented on src/helper/Redux/types/reminder.types.js:
```diff
@@ -1,5 +1,16 @@
 // TODO: types of reminder
 export const FETCH_REMINDER = 'FETCH_REMINDER';
+export const FETCH_REMINDER_SUCCESS = 'FETCH_REMINDER_SUCCESS';
+export const FETCH_REMINDER_FAILURE = 'FETCH_REMINDER_FAILURE';
+
```
also, review this status indicator I created. and trigger the value like `false` to `true` right? using reducers.

abhishekpatel946 commented on src/helper/Redux/actions/reminder.action.js:
```diff
@@ -6,9 +6,28 @@ import {
+    },
+  };
+};
+export const fetchReminderSucess = (status = false) => {
+  return {
+    type: FETCH_REMINDER,
+    payload: {
+      status: true,
+    },
+  };
+};
+export const fetchReminderFailure = (status = false) => {
+  return {
+    type: FETCH_REMINDER,
+    payload: {
+      status: true,
+    },
```
here I change the value of our status indicators, this is the good direction and data flow.

abhishekpatel946 commented on src/helper/Redux/sagas/reminder/reminder.saga.js:
```diff
@@ -1,20 +1,52 @@
     yield put(setReminder(reminders));
+    yield put(setReminderSuccess(true));
   } catch (error) {
+    yield put(setReminderFailure(true));
```
here I trigger the values `success` or `failure` indicator

abhishekpatel946 commented on src/helper/Redux/sagas/reminder/reminder.saga.js:
```diff
@@ -1,20 +1,52 @@
 // TODO: watcher saga
 export function* waitForFetchReminders() {
   yield takeLatest(FETCH_REMINDER, fetchReminder);
+  yield takeLatest(FETCH_REMINDER_SUCCESS, fetchReminderSucess);
+  yield takeLatest(FETCH_REMINDER_FAILURE, fetchReminderFailure);
```
the watcher will continues to listen the values of `success` or `failure` indicator

abhishekpatel946 reviewed (commented):
Any type of hints is acceptable, for understand where I did a mistake.

abhishekpatel946 committed:
fix: follow req, succ, fail pattern for redux-saga

abhishekpatel946 committed:
fix: fetchService using Promise

write the fetchService to fetch realTime data using Promises with saga-watcher with real-time listner

abhishekpatel946 committed:
fix: the fetching data & dispatching issue

correct the dispatch from sagas & successfully fetch data, store into source of truth and render into table, format the dateTime recieve from firestore timestamp, handling the errors when sagas & services got the inappropriate data, follow the Requst-Success-Failure pattern with @redux-toolkit and sagas
